src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -62,30 +62,33 @@
     private static final Logger LOGGER = Logger.getLogger(BaseAmazonWebServicesCredentials.class.getName());
 
     public static final int STS_CREDENTIALS_DURATION_SECONDS = 3600;
+
     private final String accessKey;
 
     private final Secret secretKey;
 
     private final String iamRoleArn;
     private final String iamMfaSerialNumber;
+    private final Integer stsTokenDuration;
 
     /**
      * Old data bound constructor. It is maintained to keep binary compatibility with clients that were using it directly.
      */
     public AWSCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id,
                               @CheckForNull String accessKey, @CheckForNull String secretKey, @CheckForNull String description) {
-        this(scope, id, accessKey, secretKey, description, null, null);
+        this(scope, id, accessKey, secretKey, description, null, null, null);
```

Comment:
We have to keep the existing constructors to maintain binary compatibility. Instead of changing constructor, add new ones (moving the `@DataBoundConstructor` to the appropriate one) or use a `@DataBoundSetter` for the new field.